2022q4/gcc.adoc

```diff
@@ -0,0 +1,25 @@
+=== GCC: Update GCC default version to version 12
+
+Links: +
+link:https://gcc.gnu.org[GCC Project] URL: link:https://gcc.gnu.org[https://gcc.gnu.org] +
+link:https://gcc.gnu.org/gcc-11/[GCC 11 release series] URL: link:https://gcc.gnu.org/gcc-11/[https://gcc.gnu.org/gcc-11/] +
+link:https://gcc.gnu.org/gcc-12/[GCC 12 release series] URL: link:https://gcc.gnu.org/gcc-12/[https://gcc.gnu.org/gcc-12/]
+
+Contact: Lorenzo Salvadore <salvadore@FreeBSD.org> +
+
+ * The GCC default version has now been updated to version 12.
+   Thank you very much to antoine@ for running the necessary exp-runs and to all the contributors, maintainers and committers involved in the process.
+   As it has been noted last quarter, for every major version of GCC FreeBSD usually waits for the release of the second minor version to update GCC default version. However next time I would like to attempt to update the default version as soon as the first minor version of GCC 13 is out. The rationale for waiting the second minor release is to wait for other operating systems (in particular Linux distributions) to find the bugs, report them, and fix them, so that FreeBSD could focus mainly on FreeBSD specific cases. But this also means that upstream software only hear from FreeBSD rarely and mostly when it concerns FreeBSD only, thus our operating system risks to be considered as minor and unimportant for them.
+   My hope is that software authors can value supporting FreeBSD more as the number of its contributions to other projects also increases. Of course I understand that this will imply more work for all ports maintainers and I will do my best to help them as much as I can.
+   link:https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=265948[https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=2659548]
+
+ * A conflict preventing the installation of multiple versions of GCC has been solved, so now lang/gcc11 and lang/gcc12 can be installed at the same time. This was particularly important for the update of the GCC default version, since a few ports have been kept to compile with GCC 11 for now.
+   Note however that at the moment only one -devel GCC port at the time can be installed on your system. This is because I have patched the standard ports only: for the -devel port I expect upstream to fix the issue soon, by using one of the patches link:https://gcc.gnu.org/bugzilla/show_bug.cgi?id=101491[submitted by a FreeBSD user] or link:https://gcc.gnu.org/pipermail/gcc-patches/2022-November/606450.html[by myself], or using some other solution.
```

Comment:
```suggestion
   Note however that at the moment only one -devel GCC port at the time can be installed on your system. This is because I have patched the standard ports only: for the -devel port I expect upstream to fix the issue soon, by using a link:https://gcc.gnu.org/bugzilla/show_bug.cgi?id=101491[patch submitted by a FreeBSD user] or link:https://gcc.gnu.org/pipermail/gcc-patches/2022-November/606450.html[my own patch], or using some other solution.
```
Making link text more meaningful for screenreaders reading a list of links outside main text.